tests/author/test_job.py

```diff
@@ -176,3 +176,69 @@ def test_stop_traversal(self):
         job = Job(root_task, arguments)
         self.assertEqual(len(job.flat_hierarchy["tasks"]), 1)
         self.assertTrue("Elements" not in " ".join([_.title for _ in job.flat_hierarchy["tasks"]]))
+
+    def test_modify(self):
+
+        root_task, arguments = tasks.TASKS_DICT.keys()[0], {"uno": [1, 2, 3], "dos": 2, "tres": 3}
+
+        for task in Plugins().tasks.values():
+            task.flags = task.Flags.PER_ELEMENT
+            task.cmd = lambda x: ["/bin/echo", "Hello World"]
+            task.tags = ["foo", "bar"]
+
+        job = Job(root_task, arguments)
+
+        def _get_attribute_listed(job, type, attribute):
+            return [_.attributeByName.get(attribute).value for _ in job.flat_hierarchy[type]]
+
+        job.modify_cmds(predicate=False, attribute="tags", value=["foobar"])
+        self.assertListEqual(
+            [["foo", "bar"], ["foo", "bar"], ["foo", "bar"]],
+            _get_attribute_listed(job, "cmds", "tags")
+        )
+
+        job.modify_cmds(predicate=True, attribute="tags", value=["foobar"])
+        self.assertListEqual(
+            [["foobar"], ["foobar"], ["foobar"]],
+            _get_attribute_listed(job, "cmds", "tags")
+        )
+
+        class Counter(object):
```

Comment:
`from collections import Counter`

Reply:
The reason for that is, that the testjob only includes identical commandtasks. Instead of having the hassle constructing a more faceted job it was way easier going this way.

From what I can see `collections.Counter`  isn't doing counts in the fashion I'd prefer here. Maybe you can elaborate your suggestions a bit more?